messaging/broker.go

```diff
@@ -1204,7 +1260,7 @@ func (r *TopicReader) Read(p []byte) (int, error) {
 		if err == ErrReaderClosed {
 			return 0, io.EOF
 		} else if err != nil {
-			return 0, fmt.Errorf("file: %s", err)
```

Comment:
I usually check specifically for marker errors and pass them through unwrapped. It's a little tedious but there's usually not a lot of those types of errors. I'm not a huge fan of fancy inspection approaches, personally. They don't feel very idomatic.